core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -57,10 +74,10 @@ export class OptimizedSsrEngine {
       ? this.currentConcurrency >= this.ssrOptions.concurrency
       : false;
 
-    const isRendering = this.renderingCache.isRendering(
-      this.getRenderingKey(request)
-    );
-
+    // if using optimizeCsrFallback, we should always start the rendering
+    const isRendering = this.ssrOptions?.optimizeCsrFallback
+      ? false
+      : this.renderingCache.isRendering(this.getRenderingKey(request));
     if (isRendering) {
```

Comment:
nitpick: we changed the semantic of this variable and `isRendering`. I'm having hard time to propose alternative name. What about about keeping `isRendering` as it was and using the config flag separately:
```ts
const isRendering = /* keep as it was */;
const reuseCurrentRendering = this.ssrOptions.reuseCurrentRendering;
if(!reuseCurrentRendering && isRendering) {
   this.log(`CSR fallback: rendering in progress (${request?.originalUrl})`);
/*...*/
}
return (
 (
   (!isRendering || reuseCurrentRendering) &&
  ...
 );
);
```

Reply:
I did some refactor in the meantime. Please let me know if you still think this is relevant.